lib/shared-sku-app/src/AppConfig/AppConfig.tsx

```diff
@@ -0,0 +1,243 @@
+
+    if (error) {
+      this.props.sdk.notifier.error(error);
+      return false;
+    }
+
+    return {
+      parameters: toExtensionParameters(this.props.parameterDefinitions, parameters),
+      targetState: selectedFieldsToTargetState(contentTypes, selectedFields)
+    };
+  };
+
+  render() {
+    return (
+      <>
+        <div className={styles.background(this.props.color)} />
+        <div className={styles.body}>
+          <Typography>
+            <Heading>About {this.props.name}</Heading>
+            <Paragraph>{this.props.description}</Paragraph>
+            <hr className={styles.splitter} />
+          </Typography>
+          {this.renderApp()}
+        </div>
+        <div className={styles.icon}>
+          <img src={this.props.logo} alt="App logo" />
+        </div>
+      </>
+    );
+  }
+
+  onParameterChange = (key: string, e: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = e.currentTarget;
+
+    this.setState(state => ({
+      ...state,
+      parameters: { ...state.parameters, [key]: value }
+    }));
+  };
+
+  onSelectedFieldsChange = (selectedFields: SelectedFields) => {
+    this.setState({ selectedFields });
+  };
+
+  renderApp() {
+    const { contentTypes, compatibleFields, selectedFields, parameters } = this.state;
+    const { parameterDefinitions } = this.props;
+    const hasConfigurationOptions = parameterDefinitions && parameterDefinitions.length > 0;
+
+    return (
+      <>
+        {hasConfigurationOptions && (
+          <Typography>
+            <Heading>Configuration</Heading>
+            <Form>
+              {parameterDefinitions.map(def => {
+                const key = `config-input-${def.id}`;
+
+                return (
+                  <TextField
+                    required={def.required}
+                    key={key}
+                    id={key}
+                    name={key}
+                    labelText={def.name}
+                    textInputProps={{
+                      width: def.type === 'Symbol' ? 'large' : 'medium',
+                      type: def.type === 'Symbol' ? 'text' : 'number',
+                      maxLength: 255
+                    }}
+                    helpText={def.description}
+                    value={parameters[def.id]}
+                    onChange={this.onParameterChange.bind(this, def.id)}
+                  />
+                );
+              })}
+            </Form>
+            <hr className={styles.splitter} />
+          </Typography>
+        )}
+        <Typography>
+          <Heading>Assign to fields</Heading>
+          {contentTypes.length > 0 ? (
+            <Paragraph>
+              This app can only be used with <strong>JSON object</strong> fields. Select which JSON
+              fields you’d like to enable for this app.
+            </Paragraph>
+          ) : (
+            <>
+              <Paragraph>
+                This app can be used only with <strong>JSON object</strong> fields.
```

Comment:
Should support `Symbol` fields, including arrays of symbols

Reply:
has been updated, file was still wip